src/BeforeMiddlewareCaller.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use DI\Bridge\Silex\Application;
+use Invoker\Reflection\CallableReflection;
+use Invoker\ParameterResolver\ResolverChain;
+use Invoker\ParameterResolver\Container\TypeHintContainerResolver;
+use Invoker\ParameterResolver\NumericArrayResolver;
+use Invoker\ParameterResolver\ParameterResolver;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Class that takes the responsibility of resolving and calling a before middleware with injection
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class BeforeMiddlewareCaller
```

Comment:
Are there any differences between `BeforeMiddlewareCaller` and `AfterMiddlewareCaller`? Maybe there can be only one class?